### PR TITLE
feat: putmask

### LINKED ISSUE #26296: putmask

### PR DESCRIPTION
**Pull request by achak98**
# PR Description

This implements the putmask function of numpy in indexing_like_operations.py in ivy's numpy frontend.

## Related Issue

Close #26296

## Checklist

- [✅] Did you add a function?
- [✅] Did you add the tests?
- [✅] Did you run your tests and are your tests passing?
- [✅] Did pre-commit not fail on any check?
- [✅] Did you follow the steps we provided?

### CODE REVIEW

File: ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py [hunk 1]
```diff
@@ -83,6 +84,25 @@ def put_along_axis(arr, indices, values, axis):
     ivy.put_along_axis(arr, indices, values, axis)
 
 
+@to_ivy_arrays_and_back
+@with_supported_dtypes(
```
Comment (KareemMAX) [inline ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py:88]: Are you sure this decorator should reference `paddle`?

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 2]
```diff
@@ -150,6 +150,49 @@ def test_numpy_put_along_axis(
...
+    test_flags,
+    frontend,
+    backend_fw,
+    fn_tree,
+    on_device,
+):
+    input_dtypes, a = dtypes_a
+    a = a[0]
+    ip_dtype = input_dtypes[0]
+    mask = helpers.np.random.choice([True, False], size=a.shape)
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:176]: I think this is better put in the decorator. You may use  `dtype_and_values` for that with dtype=`bool`

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 3]
```diff
@@ -150,6 +150,49 @@ def test_numpy_put_along_axis(
...
+    mask = mask.astype(ip_dtype)
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:177]: Is this necessary?

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 4]
```diff
@@ -150,6 +150,49 @@ def test_numpy_put_along_axis(
...
+    values_size = helpers.np.random.randint(1, a.size + 1)
+    values = helpers.np.random.randint(-100, 100, size=values_size)
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:179]: Those might be better put in the decorator

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 5]
```diff
@@ -150,6 +150,49 @@ def test_numpy_put_along_axis(
...
+    print()
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:180]: This should be removed
```suggestion
```

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 6]
```diff
@@ -150,6 +150,49 @@ def test_numpy_put_along_axis(
...
+    values = values.astype(ip_dtype)
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:181]: This is against convention, you might find another way to achieve the same behavior (maybe shared dtypes)

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hi @achak98,
I've pointed out some few comments with your PR. Can you address them please?

**Comment by Ishticode** [comment]
Thank you very much for the PR. Closing this due to lack of activity in 6 months with changes requested remaining unresolved. Please feel free to reopen if you would like to continue working on this.  :)

**Pull request closed without merging**